### PR TITLE
Add recently-idle environment scheduler in reuse mode

### PR DESCRIPTION
**Pull request by SparkSnail**

### CODE REVIEW

**Comment by J-shang** [comment]
please fix the lint

File: ts/nni_manager/training_service/reusable/gpuScheduler.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export class GpuScheduler {
 
     // private readonly machineExecutorMap: Set<TrialDetail>;
     private readonly log: Logger = getLogger();
-    private readonly policyName: SCHEDULE_POLICY_NAME = 'round-robin';
+    private readonly policyName: SCHEDULE_POLICY_NAME = 'recently-idle';
```
Comment (QuanluZhang) [inline ts/nni_manager/training_service/reusable/gpuScheduler.ts:33]: is it possible that only apply this policy for aml training service?

File: ts/nni_manager/training_service/reusable/gpuScheduler.ts [hunk 2]
```diff
@@ -190,10 +191,30 @@ export class GpuScheduler {
...
+        for (let environment of qualifiedEnvironments) {
+            if (environment.latestTrialReleasedTime > 0 && (now - environment.latestTrialReleasedTime) < minTimeInterval) {
+                selectedEnvironment = environment;
+                minTimeInterval = now - environment.latestTrialReleasedTime;
+            }
+        }
+        if (selectedEnvironment === undefined) {
+            return this.roundRobinSelect(qualifiedEnvironments, allEnvironments);
+        }
+        selectedEnvironment.latestTrialReleasedTime = -1;
```
Comment (QuanluZhang) [inline ts/nni_manager/training_service/reusable/gpuScheduler.ts:215]: BTW, one environment only runs one trial at a time?

Comment (QuanluZhang) [reply ts/nni_manager/training_service/reusable/gpuScheduler.ts:215]: if it is, then policy looks good. if not, one environment can run two trials concurrently, then this policy becomes round robin

Comment (SparkSnail) [reply ts/nni_manager/training_service/reusable/gpuScheduler.ts:215]: It could run multiple trial at one environment, why it will become round-robin?

**Review by QuanluZhang — APPROVED** [review]

**Comment by SparkSnail** [comment]
> please fix the lint

fixed.

**Review by J-shang — APPROVED** [review]

**Pull request merged**